.github/actions/go-cache/action.yml

```diff
@@ -4,6 +4,22 @@ inputs:
   key:
     description: Cache key
     required: true
+  aws-s3-bucket:
+    description: Bucket to save cache
+    required: true
+  aws-access-key-id:
+    description: Access key to AWS
+    required: true
+  aws-secret-access-key:
+    description: Secret access key to AWS
+    required: true
+  windows_go_cache_folder:
+    description: Where windows build cache is located
+    required: true
+  with-variant-keys:
+    description: Check more generic cache keys
+    required: false
+    default: "true"
```

Comment:
Instead of `restore-keys` as the workflow before, I decided to do `boolean` indication of variant keys. when non present by default it tries to restore also less scoped cache keys.